NDTensors/src/diag/diagtensor.jl

```diff
@@ -33,6 +31,19 @@ function Array(T::DiagTensor{ElT,N}) where {ElT,N}
   return Array{ElT,N}(T)
 end
 
+function DiagonalArrays.diagview(T::NonuniformDiagTensor)
+  return data(T)
+end
+
+function DiagonalArrays.diagview(T::UniformDiagTensor)
+  return array(T)
```

Comment:
What does this output?

Reply:
Oh this is a typo I meant to put
```
return diagview(array(T))
```
To force it to have the correct length. But I am not sure if that is necessary.

Reply:
We could consider returning `Fill(getdiagindex(T, 1), diaglength(T))` here if that mostly "just works" in operations where it is used.

Reply:
It doesn't look like like any of our tests were hitting this function so I think either definition would be sufficient, just let me know which you prefer!

Reply:
`diagview(array(T))` would of course technically work but is a pretty inefficient way of doing it so it would be best to avoid that if possible, the ideal thing in terms of memory would be `Fill` but if that turns out to be too tricky to get working then we could use `fill(getdiagindex(T, 1), diaglength(T))`. Again that isn't ideal and partially ruins the point of using a uniform diagonal tensor to begin with.

Reply:
If it isn't required to get tests passing then we can just not define it for the time being.

Reply:
Makes sense I removed it and we will see what happens when tests run. Thanks!

Reply:
@mtfishman The remaining tests that fail in ITensor are attempting to call `diagview` on a `UnitaryDiagTensor` so I will implement the `fill` version which you suggested here